zebra-consensus/src/transaction/tests.rs

```diff
@@ -241,3 +241,45 @@ async fn v5_transaction_is_accepted_after_nu5_activation() {
         assert_eq!(result, Ok(expected_hash));
     }
 }
+
+#[tokio::test]
+// TODO: Remove `should_panic` once the NU5 activation heights for testnet and mainnet have been
+// defined.
+#[should_panic]
+async fn transaction_is_rejected_based_on_script() {
+    let network = Network::Mainnet;
+    let blocks = zebra_test::vectors::MAINNET_BLOCKS.iter();
+
+    let state_service = service_fn(|_| async {
+        Err(Box::new(io::Error::new(
+            io::ErrorKind::Other,
+            "Pretending the UTXO was not found",
+        )) as BoxError)
+    });
```

Comment:
I'm not sure if this is the best way to test if the validation fails :/

Reply:
I'm not sure exactly what this test is trying to do, can you add a doc comment to the test function?

Maybe the check is: `v5_transaction_is_rejected_when_script_utxo_is_missing`?

If that's the case, then nothing gets rejected, instead the verifier times out waiting for the UTXO.
(The state service doesn't track spent transaction outputs, so they're not included in its responses.)

Reply:
Done, hopefully it makes the test a bit clearer. But I'm still uncertain if it's the best way. I also considered trying to create a fake script with a single opcode to reject the transaction, or creating a `known_utxo` that has an amount X and then trying to validate the transaction trying to spend `Y > X`. Do you think these alternatives are worth looking into?

Regarding the verifier hangs waiting for the UTXO, should we open a ticket to update the documentation to state that a timeout is required for correctness starting from the state service up to the transaction verifier?

Reply:
> Regarding the verifier hangs waiting for the UTXO, should we open a ticket to update the documentation to state that a timeout is required for correctness starting from the state service up to the transaction verifier?

Usually I'd just open a PR to update the documentation. But this is a tricky one, because it impacts:
- the transaction verifier (UTXOs), and everything that calls it
- the block checkpoint verifier (previous blocks), and everything that calls it
- the state block commit requests, and everything that calls them

So we should probably do it on each of the service requests - I'll open a doc PR now.